src/shared/components/DemoApp/Counter/Counter.js

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+
+import React from 'react';
+import Helmet from 'react-helmet';
+import { observable, action } from 'mobx'
+import { observer } from 'mobx-react'
+
+@observer
+class Counter extends React.Component {
+  @observable count = 0
+
+  @action inc() {
+    this.count++
+  }
+
+  @action dec() {
+    this.count--
+  }
+
+  render() {
+    return (
+      <div style={{ textAlign: 'center' }}>
+        <Helmet title="Counter"/>
+        <div>{this.count}</div>
+        <button onClick={() => this.inc()}>Inc</button>{' '}
```

Comment:
There is no need to create a new function within the "onClick" if you bind your functions in the constructor. :)

e.g.

```jsx
class Foo extends Component {
   constructor(props) {
     super(props);
     this.clicked = this.clicked.bind(this);
   }

   clicked() {
     // do stuff
   }

   render() {
     return <button onClick={this.clicked}>poop</button>;
   }
}
```